src/util/pool_timer.cpp

```diff
@@ -0,0 +1,138 @@
+//  Copyright (c) 2016 Bibek Wagle
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/exception.hpp>
+#include <hpx/runtime/applier/applier.hpp>
+#include <hpx/util/pool_timer.hpp>
+#include <hpx/util/unlock_guard.hpp>
+#include <hpx/util/bind.hpp>
+
+#include <boost/thread/locks.hpp>
+#include <boost/make_shared.hpp>
+
+#include <hpx/include/runtime.hpp>
+#include <hpx/util/io_service_pool.hpp>
+
+namespace hpx { namespace util { namespace detail
+{
+    ///////////////////////////////////////////////////////////////////////////
+    pool_timer::pool_timer()
+        : mytimer_(hpx::get_runtime().get_thread_pool("timer_pool")->get_io_service())
+    {}
+
+    pool_timer::pool_timer(util::function_nonser<bool()> const& f,
+            util::function_nonser<void()> const& on_term,
+            boost::posix_time::time_duration microsecs, std::string const& description,
+            bool pre_shutdown)
+      : f_(f), on_term_(on_term),
+        microsecs_(microsecs), description_(description),
+        pre_shutdown_(pre_shutdown), is_started_(false), first_start_(true),
+        is_terminated_(false), is_stopped_(false),
+        mytimer_(hpx::get_runtime().get_thread_pool("timer_pool")->get_io_service())
+    {}
+
+    void pool_timer::timer_handler(const boost::system::error_code &)
+    {
```

Comment:
As said before, there is no need to pass through this dummy parameter as the function is bound below anyways.